scripts/lua2dox.lua

```diff
@@ -0,0 +1,666 @@
+              thisComment = string.sub(line,1,closeSquare-1)
+              hitend = true
+
+              -- unget the tail of the line
+              -- in most cases it's empty. This may make us less efficient but
+              -- easier to program
+              inStream:ungetLine(string_trim(string.sub(line,closeSquare+2)))
+            end
+            comment = comment .. thisComment
+          end
+          if string.sub(comment,1,1)=='@' then -- it's a long magic comment
+            outStream:write('/*' .. comment .. '*/  ')
+            fn_magic = checkComment4fn(fn_magic,comment)
+          else -- discard
+            outStream:write('/* zz:' .. comment .. '*/  ')
+            fn_magic = nil
+          end
+        else
+          outStream:writeln('// zz:"' .. line .. '"')
+          fn_magic = nil
+        end
+      elseif string.find(line,'^function') or string.find(line,'^local%s+function') then
+        -- it's a function
+        local pos_fn = string.find(line,'function')
+        -- function
+        -- ....v...
+        if pos_fn then
+          -- we've got a function
+          local fn_type
+          if string.find(line,'^local%s+') then
+            fn_type = ''--'static ' -- static functions seem to be excluded
+          else
+            fn_type = ''
+          end
+          local fn = TString_removeCommentFromLine(string_trim(string.sub(line,pos_fn+8)))
+          if fn_magic then
+            fn = fn_magic
+            fn_magic = nil
+          end
+
+          if string.sub(fn,1,1)=='(' then
+            -- it's an anonymous function
+            outStream:writelnComment(line)
+          else
+            -- fn has a name, so is interesting
+
+            -- want to fix for iffy declarations
+            local open_paren = string.find(fn,'[%({]')
+            local fn0 = fn
+            if open_paren then
+              fn0 = string.sub(fn,1,open_paren-1)
+              -- we might have a missing close paren
+              if not string.find(fn,'%)') then
+                fn = fn .. ' ___MissingCloseParenHere___)'
+              end
+            end
+
+            local dot = string.find(fn0,'[%.:]')
+            if dot then -- it's a method
+              local klass = string.sub(fn,1,dot-1)
+              local method = string.sub(fn,dot+1)
+              --TCore_IO_writeln('function ' .. klass .. '::' .. method .. ftail .. '{}')
+              --TCore_IO_writeln(klass .. '::' .. method .. ftail .. '{}')
+              outStream:writeln(
+              '/*! \\memberof ' .. klass .. ' */ '
+              .. method .. '{}'
+              )
+            else
+              -- add vanilla function
+
+              outStream:writeln(fn_type .. 'function ' .. fn .. '{}')
+            end
+          end
+        else
+          this:warning(inStream:getLineNo(),'something weird here')
+        end
+        fn_magic = nil -- mustn't indavertently use it again
+      elseif string.find(line,'=%s*class%(') then
```

Comment:
@justinmk 
Do you remember what purpose this code was added to?
At present, there seems to be no code that passes this block in the Lua code in Neovim.

Reply:
No, sorry, I don't (and didn't) have a good understanding of how lua2dox really worked. Just enough to hack in a bit...

Reply:
No problem.:)  I will make a few modifications for #11276.